### DISCUSSION TITLE
No mode specified

### DISCUSSION THREAD

**Discussion by kaykhan** [folke/trouble.nvim#512] [Q&A]
Hi,

Looking for help understanding what is happening here. Ive updated to the latest version of trouble.

My config looks like

```
require("trouble").setup({})
require("trouble").next({ skip_groups = true, jump = true })
require("trouble").previous({ skip_groups = true, jump = true })

vim.keymap.set("n", "<leader>xx", "<cmd>TroubleToggle<cr>", { silent = true, noremap = true })
vim.keymap.set("n", "<leader>xw", "<cmd>TroubleToggle workspace_diagnostics<cr>", { silent = true, noremap = true })
vim.keymap.set("n", "<leader>xd", "<cmd>TroubleToggle document_diagnostics<cr>", { silent = true, noremap = true })
vim.keymap.set("n", "<leader>xl", "<cmd>TroubleToggle loclist<cr>", { silent = true, noremap = true })
vim.keymap.set("n", "<leader>xq", "<cmd>TroubleToggle quickfix<cr>", { silent = true, noremap = true })
vim.keymap.set("n", "gR", "<cmd>TroubleToggle lsp_references<cr>", { silent = true, noremap = true })
```

![image](https://github.com/folke/trouble.nvim/assets/5202527/3e9a27b9-08b2-4583-a46c-5b7b897cfb83)

**Comment by b0ae989c** [comment]
`require("trouble").next` is no longer part of Trouble v3. Should close.

**Comment by folke** [comment]
That whole config is wrong. Please just read the docs.

**Reply by Kiryoko** [reply]
Am I missing something, or has the workspace diagnostic functionality been removed in the latest version? I can only see how to open up diagnostics for the current buffer :(

**Reply by folke** [reply]
Yes, you didn't read the docs. It's in there

**Comment by sirus20x6** [comment]
also have this issue. I only see two files under docs. examples.md and filter.md. neither one say anything about what to do for the error

**Reply by b0ae989c** [reply]
See https://github.com/folke/trouble.nvim/blob/main/doc/trouble.nvim.txt

**Comment by sirus20x6** [comment]
if I copy and paste the configuration from the lazy.nvim section the error message remains

**Comment by sirus20x6** [comment]
still having the same issue. if someone knows of a fix please chime in.

**Comment by stooj** [comment]
`TroubleToggle` isn't a command (any more).

My config is in lua, which is a bit less documented, but for example my `xx` mapping is:

```
{
            action = function()
                require("trouble").toggle("diagnostics")
            end,
            key = "<leader>xx",
            mode = "",
        },
```

I believe the equivalent cmd (from the [docs](https://github.com/folke/trouble.nvim/blob/main/doc/trouble.nvim.txt#L88)) is

```
{
          "<leader>xx",
          "<cmd>Trouble diagnostics toggle<cr>",
          desc = "Diagnostics (Trouble)",
        },
```

**Reply by ffalor** [reply]
thank you @stooj, my config which has worked for a long time stopped working. I guess it was due to an upgrade.
